resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/AutoBuildJob.java

```diff
@@ -166,7 +167,6 @@ private void doBuild(IProgressMonitor monitor) throws CoreException, OperationCa
 			if (!result.isOK()) {
 				throw new ResourceException(result);
 			}
-			buildNeeded = false;
```

Comment:
This means, `buildNeeded` is now never set to false? Why that?

Reply:
hidden in shouldBuild
forceBuild = avoidBuild = buildNeeded = false;

Reply:
I have added a comment before the if to make this clearer for the future reader.